File: src/main/java/com/barbaros/expense/service/ExpenseService.java
package com.barbaros.expense.service;

import com.barbaros.expense.dto.ExpenseDTO;
import com.barbaros.expense.model.Expense;
import com.barbaros.expense.repository.ExpenseRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.stream.Collectors;

@Service
public class ExpenseService {

    private final ExpenseRepository expenseRepository;

    public ExpenseService(ExpenseRepository expenseRepository) {
        this.expenseRepository = expenseRepository;
    }

    public ExpenseDTO createExpense(ExpenseDTO expenseDTO) {
        Expense expense = new Expense(expenseDTO.getDescription(), expenseDTO.getAmount());
        Expense savedExpense = expenseRepository.save(expense);
        return new ExpenseDTO(savedExpense.getId(), savedExpense.getDescription(), savedExpense.getAmount(), savedExpense.getLabel());
    }

    public List<ExpenseDTO> getAllExpenses() {
        return expenseRepository.findAll()
                .stream()
                .map(expense -> new ExpenseDTO(expense.getId(), expense.getDescription(), expense.getAmount(), expense.getLabel()))
                .collect(Collectors.toList());
    }

    public ExpenseDTO getExpenseById(Long id) {
        return expenseRepository.findById(id)
                .map(expense -> new ExpenseDTO(expense.getId(), expense.getDescription(), expense.getAmount(), expense.getLabel()))
                .orElse(null);
    }
}

File: src/main/kotlin/com/barbaros/expense/service/ExpenseService.kt
package com.barbaros.expense.service

import com.barbaros.expense.dto.ExpenseDTO
import com.barbaros.expense.model.Expense
import com.barbaros.expense.repository.ExpenseRepository
import org.springframework.stereotype.Service

@Service
class ExpenseService(private val expenseRepository: ExpenseRepository) {

    fun createExpense(expenseDTO: ExpenseDTO): ExpenseDTO {
        val expense = Expense(
            description = expenseDTO.description,
            amount = expenseDTO.amount
        )
        val savedExpense = expenseRepository.save(expense)
        return ExpenseDTO(savedExpense.id, savedExpense.description, savedExpense.amount)
    }

    fun getAllExpenses(): List<ExpenseDTO> {
        return expenseRepository.findAll().map { expense ->
            ExpenseDTO(expense.id, expense.description, expense.amount)
        }
    }

    fun getExpenseById(id: Long): ExpenseDTO? {
        val expense = expenseRepository.findById(id).orElse(null) ?: return null
        return ExpenseDTO(expense.id, expense.description, expense.amount)
    }
}
